### PR TITLE
[CLS-4112] [CLS-7557] Natis agent injection

### PR DESCRIPTION
**Pull request by natishauli**

### CODE REVIEW

File: charts/s1-agent/templates/cluster/cluster-role.yaml [hunk 1]
```diff
@@ -14,4 +14,9 @@ rules:
 - apiGroups: ["batch", "apps"]
   resources: ["jobs", "deployments", "daemonsets", "replicasets", "statefulsets", "crontabs"]
   verbs: ["get", "watch", "list"]
+{{ if .Values.configuration.env.helper.agent_injection_enabled }}
```
Comment (danisevas) [inline charts/s1-agent/templates/cluster/cluster-role.yaml:17]: agent_injection_enabled is a long name that can be tiresome to set. can you change it to something like .Values.configuration.env.injection_enabled? even outside of the "helper" section

File: charts/s1-agent/templates/common/configmaps.yaml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    - name: agent
+{{- if .Values.secrets.imagePullSecret }}
+      imagePullSecrets:
+        - name: {{ .Values.secrets.imagePullSecret }}
+{{- end }}
+      image: "{{ include "agent.full_url" . }}"
+      imagePullPolicy: {{ default "IfNotPresent" .Values.configuration.imagePullPolicy }}
+      securityContext:
+        runAsUser: 0
+        runAsGroup: 0
```
Comment (danisevas) [inline charts/s1-agent/templates/common/configmaps.yaml:20]: why are we running as root?

**Review by danisevas — CHANGES_REQUESTED** [review]

File: charts/s1-agent/templates/common/configmaps.yaml (hunk 2 above)
Comment (natishauli) [reply charts/s1-agent/templates/common/configmaps.yaml:20]: Admin capabilities are not allowed on fargate so we must run as root to access the monitored container filesystem.

File: charts/s1-agent/templates/cluster/cluster-role.yaml (hunk 1 above)
Comment (natishauli) [reply charts/s1-agent/templates/cluster/cluster-role.yaml]: done

**Review by danisevas — APPROVED** [review]

**Pull request merged**